src/System.Globalization/tests/TextElementEnumerator/TextElementEnumeratorTests.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class TextElementEnumeratorMoveNext
+    {
+        [Fact]
+        public void Enumerate()
+        {
+            string[] expectedElements = new string[] { "\uD800\uDC00", "\uD800\uDC00", "\u0061\u0300", "\u0061\u0300", "\u00C6" };
+            int[] expectedElementIndices = new int[] { 0, 2, 4 };
+            TextElementEnumerator enumerator = StringInfo.GetTextElementEnumerator("\uD800\uDC00\u0061\u0300\u00C6");
+
+            for (int i = 0; i < 2; i++)
```

Comment:
Unrolling the loop is probably sufficient; that way the line number tells you if it was the first or second pass.

But since your first inclination was to make it a loop, add a comment :)